src/TuDa.CIMS.Web/Components/ShoppingCartFooter.razor.cs

```diff
@@ -26,4 +26,21 @@ public partial class ShoppingCartFooter() : ComponentBase
     /// The total price of the purchase
     /// </summary>
     public double TotalPrice => Purchase?.TotalPrice ?? 0.0;
+
+    /// <summary>
+    /// Opens the popup.
+    /// </summary>
+    private Task OpenDialogAsync()
+    {
+        return dialogService.ShowAsync<ShoppingCartSubmitPopup>("Custom Options Dialog",
+            new DialogOptions() { CloseButton = true });
+    }
+
+    /// <summary>
+    /// Initialize OnCheckout EventCallback so that OpenDialogAsync is called.
+    /// </summary>
+    protected override void OnInitialized()
+    {
+        OnCheckout = EventCallback.Factory.Create(this, OpenDialogAsync);
+    }
```

Comment:
🔧 I think this should be handled on a higher level. The `ShoppingCartPage` should subscribe a function to `OnCheckout,` to open the Dialog. When done like this, every information gathered by the dialog is lost.